test/test_implicit_free_surface_solver.jl

```diff
@@ -27,7 +27,18 @@ function set_simple_divergent_velocity!(model)
 
     imid = Int(floor(grid.Nx / 2)) + 1
     jmid = Int(floor(grid.Ny / 2)) + 1
-    CUDA.@allowscalar u[imid, jmid, 1] = 1
+
+    k_index = 1
+ 
+    grid isa ImmersedBoundaryGrid && begin
+        while grid.immersed_boundary.bottom_height[imid, jmid] > grid.underlying_grid.zᵃᵃᶜ[k_index]
+            k_index += 1
+        end
+    end
+
+    k_index = k_index + 1 ≤ grid.Nz ? k_index + 1 : k_index
```

Comment:
why not just having `k_index = grid.Nz`?

Reply:
that could also work given that the immersed boundary doesn't reach the surface

Reply:
see [1638f39](https://github.com/CliMA/Oceananigans.jl/pull/2723/commits/1638f392eb6f3cb04d26796d0f785f01a4f997cf)